DataStructures/BitList.cs

```diff
@@ -0,0 +1,1753 @@
+            }
+            else
+            {
+                throw new IndexOutOfRangeException();
+            }
+        }
+
+        /// <summary>
+        /// Replace a list of bit of the current instance with another list of bit.
+        /// </summary>
+        public void Replace(BitList oldBitList, BitList newBitList)
+        {
+            if (IsEmpty)
+            {
+                throw new EmptyInstanceException();
+            }
+            else
+            {
+                var thisBitListText = ToText();
+                var oldBitListText = oldBitList.ToText();
+                var newBitListText = newBitList.ToText();
+                var bitsText = thisBitListText.Replace(oldBitListText, newBitListText);
+                bits = Parse(bitsText).bits;
+            }
+        }
+
+        /// <summary>
+        /// Write the current instance into a file.
+        /// </summary>
+        public void Write(string filePath)
+        {
+            try
+            {
+                File.WriteAllBytes(filePath, ToByteArray());
+            }
+            catch
+            {
+                throw new FileWritingException();
+            }
+        }
+
+        /// <summary>
+        /// Convert this instance to a text rapresentation.
+        /// </summary>
+        public string ToText()
+        {
+            if (IsEmpty)
+            {
+                return string.Empty;
+            }
+
+            var text = string.Empty;
+            foreach (var bit in ToBooleanArray())
+            {
+                text += bit ? "1" : "0";
+            }
+
+            return text;
+        }
+
+        /// <summary>
+        /// Convert this instance to a specified unmanaged non primitive value array.
+        /// </summary>
+        /// <typeparam name="T"></typeparam>
+        /// <returns></returns>
+        public T[] ToUnmanagedNonPrimitiveValueArray<T>() where T : unmanaged
+        {
+            try
+            {
+                if (typeof(T).IsPrimitive || typeof(T).IsPointer)
+                {
+                    throw new Exception();
+                }
+                else if (IsEmpty)
+                {
+                    return new T[0];
+                }
+
+                var bytes = ToByteArray().ToList();
+
+                //Unit of measure: byte (8 bit).
+                var size = Marshal.SizeOf(typeof(T).IsEnum ? Enum.GetUnderlyingType(typeof(T)) : typeof(T));
+
+                while (bytes.Count % size != 0)
+                {
+                    //Add a missing byte.
+                    bytes.Add(0);
+                }
+
+                var result = new List<T>(); //A list of unmanaged non primitive value type objects.
+                for (var i = 0; i < bytes.Count; i += size)
+                {
+                    if (typeof(T).IsEnum)
+                    {
+                        T obj = default;
+                        switch (Enum.GetUnderlyingType(typeof(T)).Name)
```

Comment:
Codacy found an issue: [Add a 'default' clause to this 'switch' statement.](https://app.codacy.com/manual/siriak/C-Sharp/pullRequest?prid=5559704)